### ISSUE TITLE
XLForm weird layout issue in iOS 9

### ISSUE THREAD

**Issue by AtharvaVaidya** [xmartlabs/XLForm#519]
![xlform layout issue](https://cloud.githubusercontent.com/assets/9576912/9428904/f35ba09a-49db-11e5-8182-89eeec97823b.gif)

**Comment by zenkimoto** [comment]
Did you turn off "Adjust Scroll View Insets" option?  

Check out this link:  http://stackoverflow.com/questions/18924431/ios-7-navigationcontroller-is-setting-the-contentinset-and-contentoffset-of-m

Make sure you turn it on and try it again.

**Comment by AtharvaVaidya** [comment]
I had turned it on. 

I confirmed this was an issue with XLForm because I trie the same on iOS 8 and it didn't have this problem.

**Comment by MuscleRumble** [comment]
Having the same issue. @AtharvaVaidya did you solve this yet?

**Comment by matt-holden** [comment]
This Radar explains the problem: https://github.com/lionheart/openradar-mirror/issues/11

tldr; In iOS9, `automaticallyAdjustsScrollViewInsets` is not working for table views that have been created/added in code.

**Comment by matt-holden** [comment]
Adding this to `viewDidAppear` (Swift) in my `XLFormViewController` subclass produced the expected layout:

```swift
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        self.tableView.translatesAutoresizingMaskIntoConstraints = false
        self.view.addConstraint(NSLayoutConstraint(item: self.tableView!, attribute: .Top, relatedBy: .Equal, toItem: self.topLayoutGuide, attribute: .Bottom, multiplier: 1.0, constant: 0.0))
        self.view.addConstraint(NSLayoutConstraint(item: self.tableView!, attribute: .Bottom, relatedBy: .Equal, toItem: self.bottomLayoutGuide, attribute: .Top, multiplier: 1.0, constant: 0.0))
        self.view.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat("H:|[tv]|", options: NSLayoutFormatOptions(), metrics: nil, views: ["tv": self.tableView!]))
    }
```

**Comment by MuscleRumble** [comment]
Thanks @matt-holden! Seems to work. For other people that are too lazy and want the Obj-C code:

```objc
- (void)viewWillAppear:(BOOL)animated {
	[super viewWillAppear:animated];
	self.tableView.translatesAutoresizingMaskIntoConstraints = NO;
	[self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.tableView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.topLayoutGuide attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0]];
	[self.view addConstraint:[NSLayoutConstraint constraintWithItem:self.tableView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self.bottomLayoutGuide attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0]];
	[self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[tv]|" options:0 metrics:nil views:@{ @"tv": self.tableView }]];
}
```

What I actually don't understand is (and haven't checked/experimented with due to time constraints): Why does the example project of XLForm doesn't have this issue? I mean... I even use storyboard in my project... I know that XLForm creates the table view programatically nonetheless, but that's still puzzling me...

At least we have a workaround, thanks for that!

**Comment by ReDetection** [comment]
same as #483

**Comment by adrianmacneil** [comment]
See https://github.com/xmartlabs/XLForm/issues/483#issuecomment-138707904 for a simpler workaround

**Comment by MuscleRumble** [comment]
It looks simpler, but I prefer the workaround with auto layout. :D But yeah, both should work...
